### DISCUSSION TITLE
Fast refresh Next.js 13

### DISCUSSION THREAD

**Discussion by hizuvie** [vercel/next.js#49729] [App Router]
Currently i am getting a fast refresh error, when i save in dev environment, my browser running localhost it freezes, is this a bug?, i want to find a way to fix it

**Comment by Taeyooooon** [comment]
I got same issue and solved it by clearing browser cache👍

**Reply by Myroslav-mint** [reply]
I have the same issue. After any change, the browser tab became frozen until I reopened it. 

I have tried deleting `.next`, `node_modules` folders open in another browser etc., already. Anything would help me

```JSON
"dependencies": {
    "@types/node": "18.16.1",
    "@types/react": "18.2.6",
    "@types/react-dom": "18.2.4",
    "eslint": "8.40.0",
    "eslint-config-next": "13.4.2",
    "flowbite": "^1.6.5",
    "flowbite-react": "^0.4.4",
    "next": "13.4.2",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "typescript": "5.0.4"
  },```

**Reply by Myroslav-mint** [reply]
I figured it out. In my case, the root of the problem was `use client` directive for the root `page.tsx` module.

**Reply by anarkrypto** [reply]
In my case, the problem was I forgot to remove 'async' from a function in a page.tsx after add a 'use client' directive.

**Reply by simpel** [reply]
Thanks for posting this @anarkrypto. Spent the day trying to understand why my client component stalled the app only to realize I had a bad async.

**Reply by Shahzad6077** [reply]
thanks @anarkrypto 😃

**Reply by carrithers** [reply]
Yep, spent about 5 hours trying to figure out that problem. Thought I was going mad. Thanks @anarkrypto

**Reply by callmevari** [reply]
Thanks @anarkrypto! Spent hours trying to fix the freeze 👍
